rust/noosphere/src/ffi/noosphere.rs

```diff
@@ -0,0 +1,74 @@
+use anyhow::Result;
+use safer_ffi::prelude::*;
+use url::Url;
+
+use crate::noosphere::{NoosphereContext as NoosphereContextImpl, NoosphereContextConfiguration};
+
+ReprC! {
+    #[ReprC::opaque]
+    pub struct NoosphereContext {
+        inner: NoosphereContextImpl
+    }
```

Comment:
I like the FFI interface here (FFI opaque wrappers around their Rust counterparts)